test/e2e/field_selectors/field_selectors_test.go

```diff
@@ -0,0 +1,196 @@
+			Expect(out).To(ContainSubstring(fmt.Sprintf("%s/%s", deviceYAMLPath, deviceName)))
+		})
+	})
+
+	Context("apply/fleet", func() {
+		It("create for a complete fleet", func() {
+			_, _ = harness.CLI("delete", "fleet")
+			out, err := harness.CLI("apply", "-f", filepath.Join(util.GetTopLevelDir(), fleetYAMLPath))
+			Expect(err).ToNot(HaveOccurred())
+			Expect(out).To(MatchRegexp(`(200 OK|201 Created)`))
+		})
+	})
+
+	Context("Basic Functionality Tests", func() {
+
+		It("filters devices by name", func() {
+			deviceName := extractDeviceName()
+			out, err := harness.CLI("get", "devices", "--field-selector", fmt.Sprintf("metadata.name=%s", deviceName))
+			Expect(err).ToNot(HaveOccurred())
+			Expect(out).To(ContainSubstring(deviceName))
+		})
+
+		It("filters devices by owner", func() {
+			out, err := harness.CLI("get", "devices", "--field-selector", "metadata.owner=Fleet/default")
+			Expect(err).ToNot(HaveOccurred())
+			Expect(out).To(ContainSubstring("Fleet/default"))
+		})
+
+		It("filters devices by creation timestamp", func() {
+			deviceName := extractDeviceName()
+
+			startTimestamp, endTimestamp := generateTimestamps()
+			out, err := harness.CLI("get", "devices", "--field-selector",
+				fmt.Sprintf("metadata.creationTimestamp>=%s,metadata.creationTimestamp<%s", startTimestamp, endTimestamp))
+			Expect(err).ToNot(HaveOccurred())
+			Expect(out).To(ContainSubstring(deviceName))
+		})
+	})
+
+	Context("Advanced Functionality Tests", func() {
+		It("filters devices by label with contains operator", func() {
+			deviceName := extractDeviceName()
+			out, err := harness.CLI("get", "devices", "--field-selector", "metadata.labels contains region=eu-west-1")
+			Expect(err).ToNot(HaveOccurred())
+			Expect(out).To(ContainSubstring(deviceName))
+		})
+
+		It("filters devices by multiple field selectors", func() {
+			deviceName := extractDeviceName()
+			startTimestamp, _ := generateTimestamps()
+			out, err := harness.CLI("get", "devices", "--field-selector",
+				fmt.Sprintf("metadata.labels contains region=eu-west-1,metadata.creationTimestamp>=%s", startTimestamp))
+			Expect(err).ToNot(HaveOccurred())
+			Expect(out).To(ContainSubstring(deviceName))
+		})
+
+		It("filters devices by nested field", func() {
+			out, err := harness.CLI("get", "devices", "--field-selector", "status.updated.status=Unknown")
+			Expect(err).ToNot(HaveOccurred())
+			Expect(out).To(ContainSubstring("Unknown"))
+		})
+
+		It("excludes devices by name", func() {
+			out, err := harness.CLI("get", "devices", "--field-selector", "metadata.name!=device1-name")
+			Expect(err).ToNot(HaveOccurred())
+			Expect(out).ToNot(ContainSubstring("device1-name"))
+		})
+	})
+
+	Context("Negative Tests", func() {
+		It("returns an error for an invalid field selector", func() {
+			out, err := harness.CLI("get", "devices", "--field-selector", "invalid.field")
+			Expect(err).To(HaveOccurred())
+			Expect(out).To(ContainSubstring("unknown or unsupported selector"))
```

Comment:
I got a comment in my PR that we'd better create a type like "Message" in the below example instead of hardcode the returned strings, since they might change in the future, and it would be easier to maintain the code if we just change them in one place.
https://github.com/flightctl/flightctl/blob/7a9d13eef7913a247661d7672ee95cd2508275e0/test/e2e/agent/agent_suite_test.go#L17